### PR TITLE
Create claims from slices instead of scalars.

### PR DESCRIPTION
**Pull request by ParnianAlimi**
Addresses core part of CRYP-166.

### CODE REVIEW

**Comment by aafshar-poly** [comment]
@ParnianAlimi is this WIP?

**Comment by ParnianAlimi** [comment]
> @ParnianAlimi is this WIP?

Yes. This is the bulk of it, but I'll have to update the FFI as well, and also update the framework around the same time so it wouldn't break it.

**Comment by ParnianAlimi** [comment]
@aafshar-poly this is ready to review :)

File: src/claim_proofs/claim_proofs.rs [hunk 1]
```diff
@@ -57,22 +56,55 @@ lazy_static! {
...
+fn slice_to_scalar(data: &[u8]) -> Scalar {
+    use blake2::{Blake2b, Digest};
+    let mut hash = [0u8; 64];
+    hash.copy_from_slice(
+        Blake2b::default()
+            .chain(data.as_ref())
+            .finalize()
+            .as_slice(),
+    );
+    Scalar::from_bytes_mod_order_wide(&hash)
```
Comment (vkomenda) [inline src/claim_proofs/claim_proofs.rs:69]:
```suggestion
    let hash = Blake2b::digest(data);
    Scalar::from_bytes_mod_order_wide(hash.as_ref())
```
It's just a utility function but it does the same thing.

Comment (ParnianAlimi) [reply src/claim_proofs/claim_proofs.rs:69]: `Scalar::from_bytes_mod_order_wide()` expects a slice of exactly 64 bytes, that's why we have to copy the hash into a slice. I agree that in this case calling `digest()` is cleaner than `chain()`.

**Review by aafshar-poly — APPROVED** [review]
:+1:

**Pull request merged**